packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -248,6 +267,64 @@ module.exports = (log, db, config, customs, push, mailer, subhub, profile) => {
         return subhub.getCustomer(uid);
       },
     },
+    {
+      method: 'PUT',
+      path: '/oauth/subscriptions/active/{subscriptionId}',
+      options: {
+        auth: {
+          payload: false,
+          strategy: 'oauthToken',
+        },
+        validate: {
+          params: {
+            subscriptionId: validators.subscriptionsSubscriptionId.required(),
+          },
+          payload: {
+            planId: validators.subscriptionsPlanId.required(),
+          },
+        },
+      },
+      handler: async function(request) {
+        log.begin('subscriptions.updateSubscription', request);
+        if (!stripeHelper) {
+          throw error.featureNotEnabled();
+        }
+
+        const { uid, email } = await handleAuth(request.auth, true);
+
+        await customs.check(request, email, 'updateSubscription');
```

Comment:
@vbudhram is anything needed on the customs server to add new block types like this?

Reply:
@shane-tomlinson If we wanted to rate limit this, it would go in https://github.com/mozilla/fxa/blob/c64de9c201b2f221c0a337b9f45d40374f597b28/packages/fxa-customs-server/lib/actions.js#L27.

However, if this endpoint is authenticated maybe it might not need to be rate limited?

Reply:
@vbudhram If we just want the option to later to limit it, then nothing is needed though? None of the other subscription related customs.check's here have entries in the customer-server.

Reply:
> if this endpoint is authenticated

This endpoint is definitely authenticated - it's for modifying a user's subscription plan, which will require an access token over from payments-server